### PR TITLE
importer: Check whether file exist before importing (#9522)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #9522 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/9544
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/9544:release-4.0-f4be3a0bf2fa
```

---

### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/9496

Problem Summary:
If TiKV has applied one command of ingest but tidb-lightning(import-client) thinks this request has failed, it would send the same ingest-request to TiKV again, which may cause TiKV panic because the file has been ingested and could not be found.
There is many case which may cause request failed while it has been applied. Such as:
1. Network lost some packets of ingest-response.
2. The command had been copied to more than half of the members of raft-group but the original leader crashed before the entry of this command committed, then the new leader committed this entry but the client has received  fail messages by the response from the original leader.

### What is changed and how it works?

We can propose one ingest-request when all the following conditions are met:
1. There is not any pending ingest-request for this sst.
2. The current leader has applied to current term, and we can ensure that the file which we want to ingest exist.
3. The ingest file must be removed before we return response to client, so that the new request will never see the same file again.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - It may increase the duration of ingest request.
- Breaking backward compatibility

### Release note 

- No release note.

### CODE REVIEW

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -361,8 +376,14 @@ impl<W: WriteBatch + WriteBatchVecExt<RocksEngine>> ApplyContext<W> {
             exec_ctx: None,
             use_delete_range: cfg.use_delete_range,
             yield_duration: cfg.apply_yield_duration.0,
+<<<<<<< HEAD
```
Comment (kennytm) [inline components/raftstore/src/store/fsm/apply.rs:379]: Merge conflict.

Comment (kennytm) [inline components/raftstore/src/store/fsm/apply.rs:379]: Merge conflict.

Comment (Little-Wallace) [reply components/raftstore/src/store/fsm/apply.rs]: Done

**Review by NingLin-P — APPROVED** [review]
/LGTM

**Pull request merged**